### PR TITLE
Separate executor

### PR DESCRIPTION
**Pull request by Nikita-Smirnov-Exactpro**

### CODE REVIEW

File: src/main/kotlin/com/exactpro/th2/common/schema/message/impl/rabbitmq/transport/EventId.kt [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        }
+        id= ""
+        book = ""
+        scope = ""
+        timestamp = Instant.EPOCH
+    }
+
+    companion object {
+        val DEFAULT_INSTANCE: EventId = EventId()  // FIXME: do smth about its mutability
+        fun newMutable() = EventId()
```
Comment (OptimumOpium) [inline src/main/kotlin/com/exactpro/th2/common/schema/message/impl/rabbitmq/transport/EventId.kt:40]: It is better to use JvmStatic annotation for methods that we don't want to be an instance method in Java
```suggestion
        @JvmStatic
        fun newMutable() = EventId()
```

File: src/main/kotlin/com/exactpro/th2/common/schema/message/impl/rabbitmq/transport/GroupBatch.kt [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        check(groups !== DEFAULT_GROUPS) {
+            "Object can be cleaned because 'groups' is immutable"
+        }
+
+        groups.clear()
+    }
+
+    companion object {
+        val DEFAULT_GROUPS: MutableList<MessageGroup> = Collections.emptyList()
+        fun newMutable() = GroupBatch(
```
Comment (OptimumOpium) [inline src/main/kotlin/com/exactpro/th2/common/schema/message/impl/rabbitmq/transport/GroupBatch.kt:36]: Same here with JvmStatic annotation

File: src/main/kotlin/com/exactpro/th2/common/schema/message/impl/rabbitmq/transport/TransportUtils.kt [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+package com.exactpro.th2.common.schema.message.impl.rabbitmq.transport
+
+import com.exactpro.th2.common.schema.filter.strategy.impl.AbstractTh2MsgFilterStrategy.*
+import com.exactpro.th2.common.schema.filter.strategy.impl.checkFieldValue
+import com.exactpro.th2.common.schema.message.configuration.FieldFilterConfiguration
+import com.exactpro.th2.common.schema.message.configuration.RouterFilter
+import io.netty.buffer.Unpooled
+
+private typealias ProtoDirection = com.exactpro.th2.common.grpc.Direction
```
Comment (OptimumOpium) [inline src/main/kotlin/com/exactpro/th2/common/schema/message/impl/rabbitmq/transport/TransportUtils.kt:25]: You can use an import alias instead.
```
import com.exactpro.th2.common.grpc.Direction as ProtoDirection
```

File: src/test/kotlin/com/exactpro/th2/common/schema/TestJsonConfiguration.kt [hunk 4]
```diff
@@ -214,23 +229,59 @@ class TestJsonConfiguration {
...
+                    .withReflectionCacheSize(512)
+                    .configure(KotlinFeature.NullToEmptyCollection, false)
+                    .configure(KotlinFeature.NullToEmptyMap, false)
+                    .configure(KotlinFeature.NullIsSameAsDefault, false)
+                    .configure(KotlinFeature.SingletonSupport, false)
+                    .configure(KotlinFeature.StrictNullChecks, false)
+                    .build()
+            )
 
             OBJECT_MAPPER.registerModule(RoutingStrategyModule(OBJECT_MAPPER))
```
Comment (OptimumOpium) [inline src/test/kotlin/com/exactpro/th2/common/schema/TestJsonConfiguration.kt:286]: I think we could make the MAPPER filed public in AbstractCommonFactory. And use the exact same mapper here when we test the configuration deseriailzation

File: src/main/java/com/exactpro/th2/common/schema/factory/AbstractCommonFactory.java [hunk 5]
```diff
@@ -482,14 +507,8 @@ protected CradleNonConfidentialConfiguration getCradleNonConfidentialConfigurati
...
-    /**
-     * @return Schema cradle configuration
-     * @throws IllegalStateException if cannot read configuration
-     * @deprecated please use {@link #getCradleManager()}
-     */
-    @Deprecated
-    public CradleConfiguration getCradleConfiguration() {
-        return new CradleConfiguration(getCradleConfidentialConfiguration(), getCradleNonConfidentialConfiguration());
+    protected CassandraStorageSettings getCassandraStorageSettings() {
+        return getConfigurationOrLoad(CassandraStorageSettings.class, true);
```
[the remark this answers is not in the document]
Comment (Nikita-Smirnov-Exactpro) [reply src/main/java/com/exactpro/th2/common/schema/factory/AbstractCommonFactory.java:511]: I agree that using implementation class is bad idea, but splitting to modules takes a lot of time.

**Review by OptimumOpium — APPROVED** [review]

**Review by OptimumOpium — APPROVED** [review]

File: build.gradle [hunk 6]
```diff
@@ -175,18 +167,33 @@ tasks.register('integrationTest', Test) {
...
     implementation 'com.google.protobuf:protobuf-java-util'
-    implementation 'com.exactpro.th2:grpc-service-generator:3.2.2'
+    implementation 'com.exactpro.th2:grpc-service-generator:3.4.0-TH2-4887-vulnerabilities-5346040251-1bed6cd-SNAPSHOT'
     implementation "com.exactpro.th2:cradle-cassandra:${cradleVersion}"
 
+    def autoValueVersion = '1.10.1'
+    implementation "com.google.auto.value:auto-value-annotations:$autoValueVersion"
+    kapt("com.google.auto.value:auto-value:$autoValueVersion") {
+        //FIXME: Updated library because it is fat jar
+        // auto-value-1.10.1.jar/META-INF/maven/com.google.guava/guava/pom.xml (pkg:maven/com.google.guava/guava@31.1-jre, cpe:2.3:a:google:guava:31.1:*:*:*:*:*:*:*) : CVE-2023-2976, CVE-2020-8908
```
Comment (OptimumOpium) [inline build.gradle:192]: It is preprocessing. This library does not go into the final class path. We should configure the dependencyCheck correctly and exclude kapt configurations from checking

Comment (Nikita-Smirnov-Exactpro) [reply build.gradle:192]: I think we should check all by default to know about all vulnerabilities

**Review by OptimumCode — APPROVED** [review]

**Review by OptimumCode — APPROVED** [review]

**Review by OptimumCode — APPROVED** [review]

File: src/test/kotlin/com/exactpro/th2/common/schema/message/impl/rabbitmq/transport/ParsedMessageTest.kt [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+                setId(TEST_EVENT_ID)
+            }.build())
+        }.build()
+
+        val builder = message.toBuilder()
+        with(builder) {
+            assertEquals(TEST_MESSAGE_TYPE, type)
+            assertEquals(TEST_PROTOCOL, protocol)
+//        assertEquals(body, builder.bodyBuilder)
+//        assertEquals(metadata, builder.metadataBuilder())
```
Comment (OptimumCode) [inline src/test/kotlin/com/exactpro/th2/common/schema/message/impl/rabbitmq/transport/ParsedMessageTest.kt:122]: I think we should remove all commented lines

**Review by OptimumCode — APPROVED** [review]

**Review by OptimumCode — APPROVED** [review]

**Pull request merged**